File: 1291-sequential-digits/1291-sequential-digits.ts
function sequentialDigits(low: number, high: number): number[] {
    const result: number[] = [];

    for (let start = 1; start <= 9; start++) {

        let num = start;

        for (let next = start + 1; next <= 9; next++) {
            num = num * 10 + next;
            
            if (num >= low && num <= high) {
                result.push(num);
            }
        }
    }

    result.sort((a, b) => a - b);
    return result;
}

File: 1291-sequential-digits/1291-sequential-digits.rs
impl Solution {
    pub fn sequential_digits(low: i32, high: i32) -> Vec<i32> {
        let mut result = Vec::new();

        for start in 1..=9 {
            let mut num = start;

            for next in (start + 1)..=9 {
                num = num * 10 + next;
                
                if num >= low && num <= high {
                    result.push(num);
                }
            }
        }

        result.sort();
        result
    }
}